src/java/org/apache/cassandra/index/sai/disk/v1/SegmentBuilder.java

```diff
@@ -189,7 +190,11 @@ protected long addInternal(ByteBuffer term, int segmentRowId)
         @Override
         protected SegmentMetadata.ComponentMetadataMap flushInternal(IndexDescriptor indexDescriptor, IndexContext indexContext) throws IOException
         {
-            return graphIndex.writeData(indexDescriptor, indexContext, p -> p);
+            // VSTODO this is a bit of a hack. We call computeDeletedOrdinals to call computeRowIds on each
+            // VectorPostings, which will populate the rowIds field, but if we refactor the code, we could skip that.
+            var deletedOrdinals = graphIndex.computeDeletedOrdinals(p -> p);
```

Comment:
@jbellis I think this is the part of the code we were discussing on slack. We want to skip the mapping step IIUC.